src/Build/BackEnd/Components/SdkResolution/SdkResolverManifest.cs

```diff
@@ -11,8 +11,23 @@ namespace Microsoft.Build.BackEnd.SdkResolution
     /// </summary>
     internal class SdkResolverManifest
     {
+        public SdkResolverManifest()
+        {
+        }
+
+        public SdkResolverManifest(string name, string path, string namePattern)
+        {
+            Name = name;
+            Path = path;
+            NamePattern = namePattern;
+        }
+
+        internal string Name { get; set; }
+
         internal string Path { get; set; }
 
+        internal string NamePattern { get; set; }
```

Comment:
I would think about making this a `Regex` so that

1. We only carry one regex/manifest, ever, and
2. We can throw at manifest-parse time if the regex is invalid.

Reply:
Did it. I now throw internal errors on the creation (in case of compile problems) and when we apply (in case of timeout). I decided the internal exceptions should be fine, based on the thought that we provide no option to register custom resolvers on the MSBuild other than modifying its files.